client/allocrunner/networking_cni.go

```diff
@@ -114,14 +114,20 @@ func (c *cniNetworkConfigurator) Setup(ctx context.Context, alloc *structs.Alloc
 	netStatus := new(structs.AllocNetworkStatus)
 
 	if len(res.Interfaces) > 0 {
-		iface, name := func(r *cni.CNIResult) (*cni.Config, string) {
-			for i := range r.Interfaces {
-				if r.Interfaces[i].Sandbox != "" {
-					return r.Interfaces[i], i
-				}
+		// find an interface with Sandbox set, or any one of them if no
+		// interface has it set
+		var iface *cni.Config
+		var name string
+		for name, iface = range res.Interfaces {
+			if iface != nil && iface.Sandbox != "" {
+				break
 			}
-			return nil, ""
-		}(res)
+		}
+		if iface == nil {
```

Comment:
Would iface be `nil` here? My impression is the original bug is from iface never being set; here it is being set in the loop.

Reply:
Right, this isn't for protecting from the original bug, but for hopefully-unlikely situations from third-party plugins like: 

```go
res.Interfaces = map[string]*cni.Config{"foo": nil}
```

Reply:
Oooh I see, makes sense with the your comment above, thanks!